request.go

```diff
@@ -980,7 +994,20 @@ func (r *Request) sendRequest() *http.Response {
 		return nil
 	}
 
-	resp, err := r.config.Client.Do(r.http)
+	var err error
+	var resp *http.Response
+	if r.maxRetry == 0 {
+		resp, err = r.config.Client.Do(r.http)
+	} else {
+		for i := 0; i < r.maxRetry; i++ {
+			resp, err = r.config.Client.Do(r.http)
+			if err != nil {
+				time.Sleep(r.retryDelay)
+				continue
+			}
```

Comment:
It usually makes sense to retry only temporary network errors, like:

```
if nerr, ok := err.(net.Error); ok && nerr.Temporary() {
```

Also, it makes sense to retry timeout errors (err.Timeout()), I don't remember whether timeout error is considered temporary or not.

In addition, it usually makes sense to retry some of HTTP status codes, likely all 5xx errors except 501. However, for httpexpect it doesn't always make sense, so probably we need a separate option for this.

I guess we need to allow the user to specify something like "retry policy" (what to retry and how).

Reply:
Not necessarily. Please check at my use-case in the description. It's not simply a transient retry.

Reply:
I see. But I guess this is not the most common use case, so maybe handle it as a special retry policy?

E.g. we could have the following policies:

* retry temporary network errors
* retry temporary network errors and temporary server errors (5xx) - likely should be default when retrying is enabled
* retry any errors

What exact error are you retrying in your case?